v2/internal/controllers/crd_apimanagement_20220801_test.go

```diff
@@ -81,37 +88,86 @@ func Test_ApiManagement_20220801_CRUD(t *testing.T) {
 				APIM_NamedValue_CRUD(tc, &service)
 			},
 		},
+		testcommon.Subtest{
+			Name: "APIM Policy Fragment CRUD",
+			Test: func(tc *testcommon.KubePerTestContext) {
+				APIM_PolicyFragment_CRUD(tc, &service)
+			},
+		},
+		testcommon.Subtest{
+			Name: "APIM Policy CRUD",
+			Test: func(tc *testcommon.KubePerTestContext) {
+				APIM_Policy_CRUD(tc, &service)
+			},
+		},
+		// TODO: https://github.com/Azure/azure-service-operator/issues/3408
+		// testcommon.Subtest{
+		//      Name: "APIM Product CRUD",
+		//      Test: func(tc *testcommon.KubePerTestContext) {
+		//              APIM_Product_CRUD(tc, &service)
+		//      },
+		// },
+		testcommon.Subtest{
+			Name: "APIM Api CRUD",
+			Test: func(tc *testcommon.KubePerTestContext) {
+				APIM_Api_CRUD(tc, &service)
+			},
+		},
 	)
 }
 
 func APIM_Subscription_CRUD(tc *testcommon.KubePerTestContext, service client.Object) {
-	// Put this subscription on all APIs
+	// Create a subscription for all the apis
 	subscription := apim.Subscription{
-		ObjectMeta: tc.MakeObjectMetaWithName(tc.Namer.GenerateName("sub")),
+		ObjectMeta: tc.MakeObjectMetaWithName(tc.Namer.GenerateName("sub1")),
 		Spec: apim.Service_Subscription_Spec{
-			DisplayName: to.Ptr("test_subscription"),
+			DisplayName: to.Ptr("Subscription for all APIs"),
 			Scope:       to.Ptr("/apis"),
 			Owner:       testcommon.AsOwner(service),
 		},
 	}
 
-	tc.T.Log("creating apim subscription")
+	tc.T.Log("creating apim subscriptions")
 	tc.CreateResourceAndWait(&subscription)
-	defer tc.DeleteResourceAndWait(&subscription)
-
 	tc.Expect(subscription.Status).ToNot(BeNil())
+	tc.Expect(subscription.Status.Id).ToNot(BeNil())
+
+	// The subscription will have been created and it would have generated it's own
+	// PrimaryKey and SecondaryKey. Now let's see if we can set them from a Kubernetes secret.
+
+	// There should be no secrets at this point
+	secretList := &v1.SecretList{}
+	tc.ListResources(secretList, client.InNamespace(tc.Namespace))
+	tc.Expect(secretList.Items).To(HaveLen(0))
+
+	// Run sub-tests on subscription
+	tc.RunSubtests(
```

Comment:
If these tests must be run in sequence (not in parallel), best to add a comment to that effect:

```suggestion
	// Run sub-tests serially on subscription
	tc.RunSubtests(
```

We're constantly trying to keep our CI runs as quick as practical, and a common maintenance task is to make tests run in parallel where possible.

Reply:
Added a comment to all of the occurrences of this pattern